src/core/server/elasticsearch/elasticsearch_client_config.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ConfigOptions } from 'elasticsearch';
+import { readFileSync } from 'fs';
+import { Duration } from 'moment';
+import { checkServerIdentity } from 'tls';
+import url from 'url';
+import util from 'util';
+import { pick } from '../../utils';
+import { Logger } from '../logging';
+import { ElasticsearchConfig } from './elasticsearch_config';
+
+/**
+ * Config that consumers can pass to the Elasticsearch JS client is complex and includes
+ * not only entries from standard `elasticsearch.*` yaml config, but also some Elasticsearch JS
+ * client specific options like `keepAlive` or `plugins` (that eventually will be deprecated).
+ */
+export type ElasticsearchClientConfig = Pick<ConfigOptions, 'keepAlive' | 'log' | 'plugins'> &
+  Pick<
+    ElasticsearchConfig,
+    | 'apiVersion'
+    | 'customHeaders'
+    | 'logQueries'
+    | 'requestHeadersWhitelist'
+    | 'sniffOnStart'
+    | 'sniffOnConnectionFault'
+    | 'hosts'
+    | 'username'
+    | 'password'
+  > & {
+    pingTimeout?: ElasticsearchConfig['pingTimeout'] | ConfigOptions['pingTimeout'];
+    requestTimeout?: ElasticsearchConfig['requestTimeout'] | ConfigOptions['requestTimeout'];
+    sniffInterval?: ElasticsearchConfig['sniffInterval'] | ConfigOptions['sniffInterval'];
+    ssl?: Partial<ElasticsearchConfig['ssl']>;
+    loggerContext?: string;
```

Comment:
@elastic/stack-monitoring how important is it for monitoring to have `monitoring-ui` logging tag instead of just `monitoring` (https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/common/constants.js#L12) for *elastic search cluster client*? Right now monitoring is the only plugin that uses logging tag that is different from the cluster name it creates (via `plugins.elasticsearch.createCluster('monitoring')`) and we need to support additional property in ES service just for that. Would you be open to using just `monitoring` logging tag instead?

Reply:
I think we can change it? @pickypg any historical insight here?

Reply:
I don't even know why we used `monitoring-ui` versus `monitoring`. So it seems safe to me.

Reply:
Great, do you need our help with that work @azasypkin?

Reply:
> Great, do you need our help with that work @azasypkin?

@chrisronline, I'm going to just change value of [LOGGING_TAG](https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/common/constants.js#L12) to `monitoring` and update tests that depend on that. If that's fine for you then I don't need anything apart from review for the monitoring part once PR is ready :)

Reply:
Perfect. Sounds good!